be/src/vec/data_types/serde/data_type_hll_serde.cpp

```diff
@@ -102,30 +102,38 @@ void DataTypeHLLSerDe::write_column_to_arrow(const IColumn& column, const UInt8*
 }
 
 template <bool is_binary_format>
-Status DataTypeHLLSerDe::_write_column_to_mysql(
-        const IColumn& column, bool return_object_data_as_binary,
-        std::vector<MysqlRowBuffer<is_binary_format>>& result, int row_idx, int start, int end,
-        bool col_const) const {
+Status DataTypeHLLSerDe::_write_column_to_mysql(const IColumn& column,
+                                                MysqlRowBuffer<is_binary_format>& result,
+                                                int row_idx, bool col_const) const {
     auto& data_column = assert_cast<const ColumnHLL&>(column);
-    int buf_ret = 0;
-    for (ssize_t i = start; i < end; ++i) {
-        if (0 != buf_ret) {
+    if constexpr (is_binary_format) {
```

Comment:
is_binary_format can not present return_object_data_as_binary  , just because we have a switch return_object_data_as_binary,  so there is a situation with return_object_data_as_binary=true , but is_binary_format is false.

Reply:
Yes, thanks for your comment.